src/Conversion/ONNXToKrnl/Tensor/Gather.cpp

```diff
@@ -0,0 +1,117 @@
+//===----------------Gather.cpp - Lowering Gather Op----------------------=== //
+//
+// Copyright 2020 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the ONNX Gather Operator to Krnl dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp"
+
+using namespace mlir;
+
+struct ONNXGatherOpLowering : public ConversionPattern {
+  ONNXGatherOpLowering(MLIRContext *ctx)
+      : ConversionPattern(mlir::ONNXGatherOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    ONNXGatherOpAdaptor operandAdaptor(operands);
+    ONNXGatherOp gatherOp = llvm::dyn_cast<ONNXGatherOp>(op);
```

Comment:
use cast?

Reply:
thanks; I saw many other examples in the ONNXToKrnl, someone should probably do a pass